state-network.md

```diff
@@ -125,61 +83,184 @@ A node should track their own radius value and provide this value in all Ping or
 
 ### Data Types
 
-#### Component Data Elements
+#### OFFER/ACCEPT vs FINDCONTENT/FOUNDCONTENT payloads
+
+The data payloads for many content types in the history network differ between OFFER/ACCEPT and FINDCONTENT/FOUNDCONTENT.
+
+The OFFER/ACCEPT payloads need to be provable by their recipients.  These proofs are useful during OFFER/ACCEPT because they verify that the offered data is indeed part of the canonical chain.
+
+The FINDCONTENT/FOUNDCONTENT payloads do not contain proofs because a piece of state can exist under many different state roots.  All payloads can still be proved to be the correct requested data, however, it is the responsibility of the requesting party to anchor the returned data as canonical chain state data.
+
+
+#### Helper Data Types
+
+##### Paths (Nibbles)
+
+A naive approach to storage of trie nodes would be to simply use the `node_hash` value of the trie node for storage.  This scheme however results in stored data not being tied in any direct way to it's location in the trie.  In a situation where a participant in the DHT wished to re-gossip data that they have stored, they would need to reconstruct a valid trie proof for that data in order to construct the appropriate OFFER/ACCEPT payload.  We include the `path` metadata in state network content keys so that it is possible to reconstruct this proof.
+
+We define path as a sequences of "nibbles" which represent the path through the merkle patritia trie (MPT) to reach the trie node.
+
+```
+nibble     := {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, a, b, c, d, e, f}
+NibblePair := Byte  # 2 nibbles tightly packed into a single byte
```

Comment:
Can we specify how they are packed? Meaning, does first nibble goes to lower or higher 4 bits?

We would be aligned with MPT spec if first nibble goes to high bits: [link](https://ethereum.org/developers/docs/data-structures-and-encoding/patricia-merkle-trie#specification)